### PR TITLE
Add metadata to list items

### LINKED ISSUE #68: Add information about static password to List
See https://github.com/Nitrokey/trussed-secrets-app/pull/63:

To be able to provide the appropriate actions to the user, it would be good to know if there is a static password for a credential when listing it.

### PR DESCRIPTION
**Pull request by szszszsz**
Add metadata byte for the List command.

List command now accepts a single byte arguments, which denotes the requested encoding format for the list items.

Fixes #68 

Connected: https://github.com/Nitrokey/pynitrokey/pull/391

### CODE REVIEW

File: src/authenticator.rs [hunk 1]
```diff
@@ -451,11 +451,29 @@ where
...
+                reply.push((credential.label.len() + 1) as u8)?;
+                reply.push(oath::combine(credential.kind, credential.algorithm))?;
+                reply.extend_from_slice(&credential.label).map_err(|_| 0)?;
+            }
+            _ => {
+                // Unhandled version requested
+                return Err(1);
+            }
+        }
+
```
[the remark this answers is not in the document]
Comment (szszszsz) [reply src/authenticator.rs:476]: I think YAGNI. We can change TLV to CBOR later if needed.
Also, the default `0` format is already in use. Introducing multiple tag iDs will not be backwards compatible AFAIK.

File: src/credential.rs [hunk 2]
```diff
@@ -74,6 +74,22 @@ impl Default for CredentialFlat {
...
+        res |= if self.encryption_key_type.unwrap() == EncryptionKeyType::PinBased {
+            1 << 1
+        } else {
+            0
+        };
+        res |= if self.login.is_some() || self.password.is_some() {
+            1 << 2
+        } else {
+            0
+        };
```
Comment (sosthene-nitrokey) [inline src/credential.rs:89]: Rather than bit manipulation I'd use the bitflags crate but this works.

Comment (szszszsz) [reply src/credential.rs:89]: I was wondering about it, but did not want to introduce another dependency (I've missed it is used). Will keep in mind to migrate to it on the next occasion.

File: src/authenticator.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (szszszsz) [reply src/authenticator.rs:476]: Marking as resolved for now

**Pull request closed without merging**